sdk/dotnet/Pulumi.Automation.Tests/LocalWorkspaceTests.cs

```diff
@@ -826,5 +827,34 @@ static void AssertUpResult(UpResult upResult, string value)
                 Assert.True(expSecretValue.IsSecret);
             }
         }
+    
+        [Fact]
+        public async Task PulumiVersionTest()
+        {
+            using var workspace = await LocalWorkspace.CreateAsync();
+            Assert.Matches("(\\d+\\.)(\\d+\\.)(\\d+)(-.*)?", workspace.PulumiVersion.ToString());
+        }
+
+        [Theory]
+        [InlineData("100.0.0", true)]
+        [InlineData("1.0.0", true)]
+        [InlineData("2.22.0", true)]
+        [InlineData("2.1.0", false)]
+        [InlineData("2.21.2", true)]
+        [InlineData("2.21.1", false)]
+        [InlineData("2.21.0", false)]
+        public void ValidVersionTheory(string minVersion, bool errorExpected)
+        {
+            var testCurrentVersion = SemVersion.Parse("2.21.1");
```

Comment:
Curious - why do you have a fixed current version and varying min version? It would feel more natural to have it similar to the actual situation: fix the min version and test with many current versions.

Reply:
I'm going to leave this as is because I've tested it thoroughly and no one else had a problem with it in any of the other languages :)